lib/krikri/parser.rb

```diff
@@ -103,5 +103,75 @@ def get_child_nodes(name)
         raise NotImplementedError, "Can't access property #{name}"
       end
     end
+
+    ##
+    # A specialized Array object for containing Parser::Values. Provides methods
+    # for accessing and filtering values that can be chained.
+    #
+    #     my_value_array.field('dc:creator', 'foaf:name')
+    #       .match_attribute('first_name').values
+    #
+    # Methods defined on this class should return another ValueArray, an Array
+    # of literal values (retrieved from Parser::Value#value), or a single
+    # literal value.
+    class ValueArray < Array
```

Comment:
As discussed in [Slack](https://dpla.slack.com/archives/tech/p1422394042000201), `Krikri::Parser::ValueArray` should probably not subclass `Array`, based on undesirable side effects. I'll take an attempt at another pass at this using a delegator pattern. 

We should fix this before merging; I'll give it a shot by this evening.